cmd/thv-operator/api/v1alpha1/mcpregistry_types.go

```diff
@@ -187,6 +201,66 @@ type TagFilter struct {
 	Exclude []string `json:"exclude,omitempty"`
 }
 
+// MCPRegistryDatabaseConfig defines PostgreSQL database configuration for the registry API server.
+// Uses a two-user security model: separate users for operations and migrations.
+type MCPRegistryDatabaseConfig struct {
+	// Host is the database server hostname
+	// +kubebuilder:default="postgres"
+	// +optional
+	Host string `json:"host,omitempty"`
+
+	// Port is the database server port
+	// +kubebuilder:default=5432
+	// +kubebuilder:validation:Minimum=1
+	// +kubebuilder:validation:Maximum=65535
+	// +optional
+	Port int `json:"port,omitempty"`
+
+	// User is the application user (limited privileges: SELECT, INSERT, UPDATE, DELETE)
+	// Credentials should be provided via pgpass file or environment variables
+	// +kubebuilder:default="db_app"
+	// +optional
+	User string `json:"user,omitempty"`
+
+	// MigrationUser is the migration user (elevated privileges: CREATE, ALTER, DROP)
+	// Used for running database schema migrations
+	// Credentials should be provided via pgpass file or environment variables
+	// +kubebuilder:default="db_migrator"
+	// +optional
+	MigrationUser string `json:"migrationUser,omitempty"`
+
+	// Database is the database name
+	// +kubebuilder:default="registry"
+	// +optional
+	Database string `json:"database,omitempty"`
+
+	// SSLMode is the SSL mode for the connection
+	// Valid values: disable, allow, prefer, require, verify-ca, verify-full
+	// +kubebuilder:validation:Enum=disable;allow;prefer;require;verify-ca;verify-full
+	// +kubebuilder:default="disable"
```

Comment:
umm do we have plans for a saner default? :-)

Reply:
You know what, yes, I'll change it to `prefer` now. I saw this and remembered some scars from an on-prem TLS Postgres setup where we did `verify-full` which was as painful as you can imagine. So I naturally went to the complete opposite extreme for the default our of pure kneejerk reaction to the painful memories. However, I am happy to change this to `prefer` to match the default for Postgres because a user can always set it to disable if they prefered.